src/physics_swe.c

```diff
@@ -374,10 +375,12 @@ static PetscErrorCode RHSFunctionForBoundaryEdges(RDy rdy, Vec F, AMaxDiagnostic
         PetscReal hl = x_ptr[icell * ndof + 0];
 
         if (!(hl < tiny_h)) {
-          if (amax_vec_bnd[e] > amax_diags->max_wave_speed) {
-            amax_diags->max_wave_speed = amax_vec_bnd[e];
-            amax_diags->global_edge_id = edges->global_ids[e];
+          PetscReal cnum = amax_vec_bnd[icell] * edge_len / cell_area * rdy->dt;
+          if (cnum > courant_num_diags->max_courant_num) {
+            courant_num_diags->max_courant_num = cnum;
```

Comment:
This is based on https://github.com/RDycore/toy-problems/blob/main/swe_roe/ex2b.c#L1792